### DISCUSSION TITLE
Change center position of `map.Map` in real-time.

### DISCUSSION THREAD

**Discussion by hi-one-gg** [flet-dev/flet#4370] [Q&A] [answered]
### Duplicate Check

- [X] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the requested feature

I want to programmatically change the center position of `map.Map` in real-time.

I changed the `initial_center` value in the `map.Map.configuration` and called `update()`, but the center position of the map did not move and remains the same.

Is there a way to change the center of the map in real-time? If not, I would like to request this feature.

**Comment by Siva-Rajan-R** [comment]
I also faced this issue

**Comment by ndonkoHenri** [comment, marked as answer]
It is already possible, but only in latest pre-release for now: `pip install flet --pre -U`

Lot of new methods have been added. For your use case, the following will be useful:
- [`Map.center_on()`](https://github.com/flet-dev/flet/blob/5fb877b3a3f886f3475cd8ebca1cee52472d0ef7/sdk/python/packages/flet/src/flet/core/map/map.py#L250)
- [`Map.move_to()`](https://github.com/flet-dev/flet/blob/5fb877b3a3f886f3475cd8ebca1cee52472d0ef7/sdk/python/packages/flet/src/flet/core/map/map.py#L225)

Note that these methods (and equally all other events previously present in `MapConfiguration`) are part of `Map`.

**Reply by hi-one-gg** [reply]
That's an amazing answer. You really saved me, thank you so much.
If I could ask for one mo

https://github.com/user-attachments/assets/e17e3df9-94ca-454c-a367-c7a4ce6d9f0f

re thing... could you please include that feature in the official version instead of just the trial version?
